### PR TITLE
Added support for pyflyte serialize fast register

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Currently flytectl doesn't replace `{{ .dest_dir }}` in fast resgister, We are only replacing `{{ .remote_package_path }}`

remote_package_path and dest_dir aren't interchangeable though, they are distinct variables. we should be substituting both as we do in flyte-cli: https://github.com/flyteorg/flytekit/blob/master/flytekit/clis/flyte_cli/main.py#L1998,L2006

Testing 

Setup
```
pyflyte init demo && cd demo
flytectl sandbox start --source 
flytectl sandbox exec -- docker build -t myapp:v1 .
```

Pyflyte package
```
pyflyte --pkgs flyte  package --image myapp:v1   --force --fast 
 flytectl go run main.go register files temp/demo/flyte-package.tar.tz -p flytesnacks -d staging --version 11
 -------------------------------------------------------- --------- ------------------------------ 
| NAME                                                   | STATUS  | ADDITIONAL INFO              |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/0_flyte.workflows.example.say_hello_1.pb | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/1_flyte.workflows.example.my_wf_2.pb     | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/2_flyte.workflows.example.my_wf_3.pb     | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
3 rows
```
Screenshot 2021-12-24 at 4 53 55 PM

Pyflyte serialize
```
export FLYTE_INTERNAL_IMAGE=myapp: v1
pyflyte --config flyte.config serialize fast workflows -f _pb
 flytectl go run main.go register files temp/demo/_pb/* -p flytesnacks -d staging --version 12
 -------------------------------------------------------- --------- ------------------------------ 
| NAME                                                   | STATUS  | ADDITIONAL INFO              |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/0_flyte.workflows.example.say_hello_1.pb | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/1_flyte.workflows.example.my_wf_2.pb     | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
| temp/demo/_pb/2_flyte.workflows.example.my_wf_3.pb     | Success | Successfully registered file |
 -------------------------------------------------------- --------- ------------------------------ 
3 rows
```
Screenshot 2021-12-24 at 4 51 49 PM

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1984

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: cmd/register/register_util.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const registrationVersionPattern = "{{ registration.version }}"
 
 // Additional variable define in fast serialized proto that needs to be replace in registration time
 const registrationRemotePackagePattern = "{{ .remote_package_path }}"
+const registrationDestDirPattern = "{{ .dest_dir }}"
```
Comment (pmahindrakar-oss) [inline cmd/register/register_util.go:48]: Can we check if we need a separate flag for this and how the user is going to override it.
Also add examples in the doc in what cases do they use this new flag in combination with remote_package_path

Comment (evalsocket) [reply cmd/register/register_util.go:48]: Added flags for destinationDir

**Review by EngHabu — APPROVED** [review]
